lib/integrations/awsoidc/tags/tags.go

```diff
@@ -126,3 +128,40 @@ func (d AWSTags) ToIAMTags() []iamTypes.Tag {
 	}
 	return iamTags
 }
+
+// ToS3Tags returns the default tags using the expected type for S3 resources: [s3types.Tag]
+func (d AWSTags) ToS3Tags() []s3types.Tag {
+	s3Tags := make([]s3types.Tag, 0, len(d))
+	for k, v := range d {
+		k, v := k, v
+		s3Tags = append(s3Tags, s3types.Tag{
+			Key:   &k,
+			Value: &v,
+		})
+	}
+	return s3Tags
+}
+
+// ToAthenaTags returns the default tags using the expected type for Athena resources: [athenatypes.Tag]
+func (d AWSTags) ToAthenaTags() []athenatypes.Tag {
+	athenaTags := make([]athenatypes.Tag, 0, len(d))
+	for k, v := range d {
+		k, v := k, v
+		athenaTags = append(athenaTags, athenatypes.Tag{
+			Key:   &k,
+			Value: &v,
+		})
+	}
+	return athenaTags
+}
+
+// ToMap returns the default tags using the expected type for other aws resources.
+// Eg Glue resources
+func (d AWSTags) ToMap() map[string]string {
+	mapTags := make(map[string]string, len(d))
+	for k, v := range d {
+		k, v := k, v
+		mapTags[k] = v
+	}
+	return mapTags
```

Comment:
is this what we're doing here?
```suggestion
	return maps.Clone(d.(map[string]string))
```

Reply:
edoardo's suggestion is better